### PR TITLE
Paralellize hashing of large lists.

### PR DESCRIPTION
**Pull request by potuz**
See the benchmarks in #12639

### CODE REVIEW

**Review by terencechain — COMMENTED** [review]
is this something we want a feature flag over?

File: beacon-chain/state/stateutil/field_root_validator.go [hunk 1]
```diff
@@ -51,21 +53,45 @@ func validatorRegistryRoot(validators []*ethpb.Validator) ([32]byte, error) {
...
 	}
-	roots := make([][32]byte, 0, len(validators)*validatorFieldRoots)
-	for i := 0; i < len(validators); i++ {
+	wg := sync.WaitGroup{}
+	n := runtime.GOMAXPROCS(0)
+	rootsSize := len(validators) * validatorFieldRoots
+	groupSize := len(validators) / n
+	roots := make([][32]byte, rootsSize)
+	wg.Add(n - 1)
+	for j := 0; j < n-1; j++ {
```
Comment (terencechain) [inline beacon-chain/state/stateutil/field_root_validator.go:82]: why use `j` instead of `i` here?

File: crypto/hash/htr/hashtree.go [hunk 2]
```diff
@@ -1,17 +1,45 @@
...
+}
+
 // VectorizedSha256 takes a list of roots and hashes them using CPU
 // specific vector instructions. Depending on host machine's specific
 // hardware configuration, using this routine can lead to a significant
 // performance improvement compared to the default method of hashing
 // lists.
 func VectorizedSha256(inputList [][32]byte, outputList [][32]byte) {
-	err := gohashtree.Hash(outputList, inputList)
+	if len(inputList) < 5000 {
```
Comment (terencechain) [inline crypto/hash/htr/hashtree.go:24]: please add comment or constant for 5000

**Comment by potuz** [comment]
closing in favour of #12639

**Pull request closed without merging**